src/repository.rs

```diff
@@ -289,6 +290,8 @@ where
     }
 }
 
+/// A custom hook to be invoked before sending HTTP requests.
+pub type HttpHook = Box<for<'a> Fn(RequestBuilder<'a>) -> RequestBuilder<'a>>;
```

Comment:
I was thinking about this some more. I think the signatures should be

```rust
// non-public type
type HttpHook = Box<for<'a> Fn(&mut RequestBuilder<'a>) -> Result<RequestBuilder<'a>, Into<Error>>; 
```

And the function docs needs to say that it **must not** panic. I think this is better because I would rather consumers of this lib not introduce panicking code. On top of this, I think it would be good to wrap the call site itself in a `panic::catch_unwind` and returning `Error::Opaque("Hook panicked.".into())` to contain the panic and stop it from spreading elsewhere in the code.